File: plugin/jvm/src/main/java/com/twosigma/beaker/chart/xychart/plotitem/Crosshair.java
/*
 *  Copyright 2014 TWO SIGMA OPEN SOURCE, LLC
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package com.twosigma.beaker.chart.xychart.plotitem;

import com.twosigma.beaker.chart.Color;

/**
 * Crosshair
 *
 */
public class Crosshair {
  private StrokeType style;
  private Float width;
  private Color color;

  public Crosshair setStyle(StrokeType style) {
    this.style = style;
    return this;
  }

  public StrokeType getStyle() {
    return this.style;
  }

  public Crosshair setWidth(float width) {
    this.width = width;
    return this;
  }

  public Float getWidth() {
    return this.width;
  }

  public Crosshair setColor(Color color) {
    this.color = color;
    return this;
  }

  public Color getColor() {
    return this.color;
  }
}

File: kernel/scala/src/main/scala/com/twosigma/beakerx/scala/chart/xychart/plotitem/Crosshair.scala
/*
 *  Copyright 2017 TWO SIGMA OPEN SOURCE, LLC
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package com.twosigma.beakerx.scala.chart.xychart.plotitem

import com.twosigma.beakerx.chart.Color
import com.twosigma.beakerx.chart.xychart.plotitem.StrokeType
import com.twosigma.beakerx.scala.JavaAdapter._

class Crosshair extends com.twosigma.beakerx.chart.xychart.plotitem.Crosshair with CrosshairProperties {

  def this(color: Color, width: Float, style: StrokeType) {
    this()
    this.setColor(color)
    this.setWidth(width)
    this.setStyle(style)
  }

}

trait CrosshairProperties {
  this: com.twosigma.beakerx.chart.xychart.plotitem.Crosshair with CrosshairProperties =>
  def color: Option[Color] = Option(getColor)
  def color_=(color: Color): Unit = setColor(color)
  def color_=(color: java.awt.Color): Unit = setColor(color)

  def style: Option[StrokeType] = Option(getStyle)
  def style_=(style: StrokeType): Unit = setStyle(style)

  def width: Option[Float] = safeOption(getWidth)
  def width_=(width: Float): Unit = setWidth(width)
}
